terraform/cp4ba/variables.tf

```diff
@@ -0,0 +1,232 @@
+
+variable "ibmcloud_api_key" {
+  description = "IBM Cloud API key (https://cloud.ibm.com/docs/account?topic=account-userapikey#create_user_key)"
+}
+
+variable "cluster_name_id" {
+  default     = ""
+  description = "Enter your cluster id or name to install the Cloud Pak. Leave blank to provision a new Openshift cluster."
+}
+
+variable "entitled_registry_user_email" {
+  type = string
+  description = "Email address of the user owner of the Entitled Registry Key"
+//  validation {
+//    condition = can(regrex("^No resources found+$", var.entitled_registry_user_email))
+//    error_message = "At least one user must be available in order to proceed. Please refer to the README for the requirements and instructions. The script will now exit!"
+//  }
+}
+
+variable "iaas_classic_api_key" {}
+variable "iaas_classic_username" {}
+variable "ssh_public_key_file" {}
+variable "ssh_private_key_file" {}
+variable "classic_datacenter" {}
+
+variable "config_dir" {
+  default     = "./.kube/config"
+  description = "directory to store the kubeconfig file"
+}
+
+variable "region" {
+  description = "Region where the cluster is created"
+}
+
+variable "resource_group" {
+//  name       = "cloud-pak-sandbox-ibm"
+  description = "Resource group name where the cluster will be hosted."
+}
+
+variable "openshift_version" {
+  default     = "4.6"
+  type        = string
+  description = "Openshift version installed in the cluster"
+}
+//
+//variable "hasEntitlementKey" {
+//  type        = string
+//  description = "Do you have a Cloud Pak for Business Automation Entitlement Registry key (Yes/No, default: No):"
+//}
+
+variable "local_registry_server" {
+  description = "Enter the public image registry or route (e.g., default-route-openshift-image-registry.apps.<hostname>).\nThis is required for docker/podman login validation:"
+}
+
+variable "portworx_is_ready" {
+  type    = any
+  default = null
+}
+
+variable "entitled_registry_key" {
+  type        = string
+//  sensitive = true
+  description = "Do you have a Cloud Pak for Business Automation Entitlement Registry key? If not, Get the entitlement key from https://myibm.ibm.com/products-services/containerlibrary"
+}
+
+variable "namespace" {
+  type        = string
+  description = "namespace for cp4ba"
+}
+
+variable "project_name" {
+  description = "Enter a valid project name. Project name should not be 'openshift' or 'kube' or start with 'openshift' or 'kube'."
+  type = string
+//  validation {
+//    condition = can(regex("^kube+$", var.project_name))
+//    error_message = "Please enter a valid project name that should not be 'openshift' or 'kube' or start with 'openshift' or 'kube'."
+//  }
+}
+
+variable "platform_options" {
```

Comment:
Since we are either provisioning `roks` or expecting a `roks` cluster id, this script shouldn't be supporting any other platform type.  If a user wants to install on another type of cluster, they can provision their other cluster and just use your cp4ba example to install on their cluster.

Reply:
Ok, so I will need to edit the script.